Add token generation service for parity check

ethax-ross opened the pull request:
[Jira-3614](https://dfedigital.atlassian.net.mcas.ms/jira/software/projects/CPDLP/boards/87?assignee=unassigned%2C712020%3A5b5839dc-e4d6-4df5-9dd6-627cc49fe345&selectedIssue=CPDLP-3614)

### Context

We are creating a service to check parity between the API responses of ECF and NPQ reg, once the data has been migrated. As part of this we will need to call the endpoints as lead providers, so we will need to be able to generate tokens for each lead provider in both ECF and NPQ reg.

### Changes proposed in this pull request

- Add ECF models for ApiToken generation

Bring across the models from ECF that we need to be able to generate API tokens for ECF in NPQ registration.

I haven't improved the test coverage here, just a quick lift and shift from ECF.

- Add token generation service for ECF/NPQ

As part of the parity check we will need to generate API tokens for lead providers in ECF and NPQ reg on the fly to be able to make requests to the API as these providers.

Add service to generate tokens for each lead provider.

leandroalemao reviewed (commented):
@ethax-ross Looks good 👍 but couldn't really test locally here as ECF models are [marked as read only](https://github.com/DFE-Digital/npq-registration/blob/main/app/models/migration/ecf/base_record.rb#L9).. so am getting an exception `ActiveRecord::ReadOnlyRecord` when running `Migration::ParityCheck::TokenProvider.new.generate!` 

Also Is it possible to add something to limit these things to run only in non-production envs? Also could you please have a look at test coverage on sonarcloud? 90.7% on the new code
Tksss

ethax-ross commented:
> @ethax-ross Looks good 👍 but couldn't really test locally here as ECF models are [marked as read only](https://github.com/DFE-Digital/npq-registration/blob/main/app/models/migration/ecf/base_record.rb#L9).. so am getting an exception `ActiveRecord::ReadOnlyRecord` when running `Migration::ParityCheck::TokenProvider.new.generate!`
> 
> Also Is it possible to add something to limit these things to run only in non-production envs? Also could you please have a look at test coverage on sonarcloud? 90.7% on the new code Tksss

Good catch! I've updated to be writable so it works in non-test environments and added a flag to the npq_separation config to raise if its not in the migration environment.

On the code coverage I put a note in my commit that I've just lifted & shifted the ECF code here and not improved the coverage; I didn't think it was worth the effort as its short-lived code that has been in ECF prod for ages. Happy to be challenged on that though if you feel differently?

leandroalemao reviewed (approved):
Looking nice now tks @ethax-ross 👍 😄
